crates/oxc_linter/src/rules/eslint/no_unused_vars/diagnostic.rs

```diff
@@ -1,61 +1,95 @@
     if symbol_flags.is_function() {
-        "Function"
+        ("Function", "functions")
     } else if symbol_flags.is_class() {
-        "Class"
+        ("Class", "classes")
     } else if symbol_flags.is_interface() {
-        "Interface"
+        ("Interface", "interfaces")
     } else if symbol_flags.is_type_alias() {
-        "Type alias"
+        ("Type alias", "type aliases")
     } else if symbol_flags.is_enum() {
-        "Enum"
+        ("Enum", "enums")
     } else if symbol_flags.is_enum_member() {
-        "Enum member"
+        ("Enum member", "enum members")
     } else if symbol_flags.is_type_import() {
-        "Type"
+        ("Type", "types")
     } else if symbol_flags.is_import() {
-        "Identifier"
+        ("Identifier", "identifiers")
+    } else if symbol_flags.is_catch_variable() {
+        ("Catch parameter", "caught errors")
     } else {
-        "Variable"
+        ("Variable", "variables")
     }
 }
 
-pub fn used_ignored(symbol: &Symbol<'_, '_>) -> OxcDiagnostic {
-    let pronoun = pronoun_for_symbol(symbol.flags());
+pub fn used_ignored<R>(symbol: &Symbol<'_, '_>, pat: &IgnorePattern<R>) -> OxcDiagnostic
+where
+    R: fmt::Display,
+{
+    let (pronoun_singular, _) = pronoun_for_symbol(symbol.flags());
     let name = symbol.name();
 
-    OxcDiagnostic::warn(format!("{pronoun} '{name}' is marked as ignored but is used."))
+    let help_suffix = match pat {
+        IgnorePattern::None => ".".into(),
+        IgnorePattern::Default => {
+            name.strip_prefix('_').map_or(".".into(), |name| format!(" to '{name}'."))
+        }
+        IgnorePattern::Some(ref r) => {
+            format!(" to match the pattern /{r}/.")
+        }
+    };
+
+    OxcDiagnostic::warn(format!("{pronoun_singular} '{name}' is marked as ignored but is used."))
         .with_label(symbol.span().label(format!("'{name}' is declared here")))
-        .with_help(format!("Consider renaming this {}.", pronoun.cow_to_lowercase()))
+        .with_help(format!(
+            "Consider renaming this {}{help_suffix}",
+            pronoun_singular.cow_to_lowercase()
+        ))
 }
 /// Variable 'x' is declared but never used.
-pub fn declared(symbol: &Symbol<'_, '_>) -> OxcDiagnostic {
+pub fn declared<R>(symbol: &Symbol<'_, '_>, pat: &IgnorePattern<R>) -> OxcDiagnostic
+where
+    R: fmt::Display,
+{
     let (verb, help) = if symbol.flags().is_catch_variable() {
         ("caught", "Consider handling this error.")
     } else {
         ("declared", "Consider removing this declaration.")
     };
-    let pronoun = pronoun_for_symbol(symbol.flags());
     let name = symbol.name();
+    let (pronoun, pronoun_plural) = pronoun_for_symbol(symbol.flags());
+    let suffix = pat.diagnostic_help(pronoun_plural);
 
-    OxcDiagnostic::warn(format!("{pronoun} '{name}' is {verb} but never used."))
+    OxcDiagnostic::warn(format!("{pronoun} '{name}' is {verb} but never used.{suffix}"))
```

Comment:
Would it make more sense to put this in the help text? It seems like it would make more sense there. I wish we could supply multiple help messages.

Reply:
I tend to agree. It would be nice to support multiple help messages. This follows ESLint's message, so I'm going to leave it here for now.